controllers/common/finalizers/controller.go

```diff
@@ -51,8 +51,74 @@ type Reconciler struct {
 	Log logr.Logger
 }
 
-// Reconcile the common chaos
-func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+// Reconcile the common chaos to init the finalizer
+func (r *InitReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	obj := r.Object.DeepCopyObject().(v1alpha1.InnerObject)
+
+	if err := r.Client.Get(context.TODO(), req.NamespacedName, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info("chaos not found")
+		} else {
+			// TODO: handle this error
+			r.Log.Error(err, "unable to get chaos")
+		}
+		return ctrl.Result{}, nil
+	}
+
+	finalizers := obj.GetFinalizers()
+	shouldUpdate := false
+	if !obj.IsDeleted() {
+		if !ContainsFinalizer(obj.(metav1.Object), RecordFinalizer) {
+			r.Recorder.Event(obj, recorder.FinalizerInited{})
+			shouldUpdate = true
+			finalizers = append(obj.GetFinalizers(), RecordFinalizer)
+		}
+	}
+
+	if shouldUpdate {
+		updateError := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
There are two duplicate parts of update code in two Reconciler. Make a function here is better.

Reply:
@Andrewmatilde Fixed! PTAL again!